src/main/java/bio/terra/workspace/app/controller/GlobalExceptionHandler.java

```diff
@@ -29,11 +31,25 @@
   // -- validation exceptions - we don't control the exception raised
   @ExceptionHandler({
     MethodArgumentNotValidException.class,
+    MethodArgumentTypeMismatchException.class,
+    HttpMessageNotReadableException.class,
     IllegalArgumentException.class,
     NoHandlerFoundException.class
   })
   public ResponseEntity<ErrorReport> validationExceptionHandler(Exception ex) {
-    return buildErrorReport(ex, HttpStatus.BAD_REQUEST, null);
+    logger.error("Global exception handler: catch stack", ex);
+    // For security reasons, we generally don't want to include the user's invalid (and potentially
+    // malicious) input in the error response, which also means we don't include the full exception.
+    // Instead, we return a generic error message about input validation.
+    String validationErrorMessage =
+        "Request could not be parsed or was invalid: "
+            + ex.getClass().getSimpleName()
+            + ". Ensure that all types are correct and that enums have valid values.";
+    ErrorReport errorReport =
+        new ErrorReport()
+            .message(validationErrorMessage)
+            .statusCode(HttpStatus.BAD_REQUEST.value());
+    return new ResponseEntity<>(errorReport, HttpStatus.BAD_REQUEST);
```

Comment:
this is such a balance between security and user-friendliness. If possible, let's try to at least output the *name* of the variable that failed validation, e.g. "SnapshotId value could not be parsed or was invalid". I realize this won't always be possible, and would need to be captured at the point of validation and passed to this handler (as a field in the various Exceptions?). No need to hold this PR for that, but let's try as we continue to develop.

Reply:
Agreed, I'm a little concerned that we're packing too many types of errors into a general "your request is bad" message.